include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,326 @@
+ * \ingroup cigar
+ * \tparam cigar_type Must be of type seqan3::cigar_op.
+ * \tparam count_type Type of the count value; must model std::UnsignedIntegral.
+ *
+ * This alphabet can be used to represent an alignment in a compressed way using
+ * the CIGAR string representation. The seqan3::cigar alphabet implements the
+ * CIGAR elements in this context consisting of a CIGAR operator (seqan3::cigar_op)
+ * and a count value.
+ *
+ * Example usage:
+ * \snippet test/snippet/alphabet/cigar/cigar_op.cpp general
+ *
+ * \note Usually you do not want to manipulate cigar elements and vectors on
+ *       your own but convert an alignment to a cigar and back. See
+ *       seqan3::get_cigar_vector for how to convert two aligned sequences into
+ *       an cigar_vector.
+ */
+template<typename cigar_type = cigar_op, typename count_type = uint32_t>
+//!\cond
+    requires std::Same<cigar_type, cigar_op> && std::UnsignedIntegral<count_type>
+//!\endcond
+class cigar :
+    public cartesian_composition<cigar<cigar_type, count_type>, cigar_type, count_type>
+{
+private:
+    //!\brief The base type.
+    using base_type = cartesian_composition<cigar, cigar_op, uint32_t>;
+
+public:
+
+    //!\brief Equals the char_type of cigar_op type.
+    using char_type = underlying_char_t<cigar_op>;
+
+    //!\brief The type combining both types into a streamable string.
+    using string_type = std::string;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    cigar() = default;
+    constexpr cigar(cigar const &) = default;
+    constexpr cigar(cigar &&) = default;
+    constexpr cigar & operator =(cigar const &) = default;
+    constexpr cigar & operator =(cigar &&) = default;
+    ~cigar() = default;
+
+    using base_type::base_type; // Inherit non-default constructors
+
+    using base_type::operator=; // Inherit non-default assignment operators
+
+    //!\copydoc cartesian_composition::cartesian_composition(component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar(component_type const alph) {} ))
+    //!\copydoc cartesian_composition::cartesian_composition(indirect_component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar(indirect_component_type const alph) {} ))
+    //!\copydoc cartesian_composition::operator=(component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar & operator=(component_type const alph) {} ))
+    //!\copydoc cartesian_composition::operator=(indirect_component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar & operator=(indirect_component_type const alph) {} ))
+    //!\}
+
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Assign from a character. This modifies the internal sequence letter.
+    constexpr cigar & assign_char(char_type const c)
+    {
+        seqan3::assign_char(get<0>(*this), c);
+        return *this;
+    }
+
+    //!\brief Assign from a count value. This modifies the internal quality letter.
```

Comment:
_quality letter_ is a copy'n'paste error. Below there is at least two more of this (please do a grep on _quality_).